### ISSUE TITLE
Add setSWSPIdelay()

### ISSUE THREAD

**Issue by RobTillaart** [RobTillaart/MAX31855_RT#23]
- for tuning SW SPI signal timing e.g. long wires.
-

**Comment by RobTillaart** [comment]
like this
```cpp
  else  // Software SPI
  {
    digitalWrite(_select, LOW);
    for (int8_t i = 31; i >= 0; i--)
    {
      _rawData <<= 1;
      digitalWrite(_clock, LOW);
      if (swdel > 0 )  delayMicroseconds(swdel / 2 );         <<<<<<<<<
      if ( digitalRead(_miso) ) _rawData++;
      digitalWrite(_clock, HIGH);
      if (swdel > 0 ) delayMicroseconds(swdel - swdel / 2 );     <<<<<<<<<<<
    }
    digitalWrite(_select, HIGH);
  }
```
and then add  **void setSWSPIdelay(uint16_t swdel)**  and a **uint16_t getSWSPIdelay()**

this would allow adjusting timing when needed e.g. when long wires are used.

**Comment by jpsminix** [comment]
It would be nice to adjust. But i dont understand the second if (swdel > 0)... "swdel - swdel / 2" == 0/2 ??

**Comment by RobTillaart** [comment]
That is to divide the delay equally over the high and low part of the click pulse.
Testing for zero prevents an unnecessary call to delaymicroseconds.

The subtraction is for odd numbers. So if one adds up the actual delays one will get the delay set.

**Comment by RobTillaart** [comment]
merged in 0.4.1
